src/librustc_metadata/encoder.rs

```diff
@@ -401,8 +401,12 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             }
         }
 
+        let repr_options = *self.tcx.item_type(adt_def_id).get_repr().expect(
+            &format!("{:?} is not an ADT", def_id)
```

Comment:
Probably not the best to have this allocating here. If you make an encoding-specific helper method you can just use `bug!` in cases other than `TyAdt`.